client/my-sites/checkout/composite-checkout/hooks/use-memo-compare.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { useRef, useEffect } from 'react';
+
+// See https://usehooks.com/useMemoCompare/
+export default function useMemoCompare< A, B >(
```

Comment:
This seems general purpose enough to warrant being put in a less specific place, but I'm not sure where that would be.

Reply:
Yeah, same. There's a few things in this directory and `lib` that fall into a general utility category but I figure we can always move them later.